### PR TITLE
Implement window/showDocument server request

### PR DESCRIPTION
**Pull request by rwols**
Implements https://github.com/microsoft/language-server-protocol/commit/ec377055d48d1bc79bbee6ad89c23ec428059f8f

### CODE REVIEW

**Comment by rwols** [comment]
I haven't actually tested this, but it might be a good basis to start out with.

**Comment by ayoub-benali** [comment]
Really cool, Metals have a custom request that does the same thing. Created an issue https://github.com/scalameta/metals-feature-requests/issues/164 to migrate it once the 3.16 spec is official.

File: plugin/core/sessions.py [hunk 1]
```diff
@@ -236,12 +238,16 @@ def get_initialize_params(variables: Dict[str, str], workspace_folders: List[Wor
...
+            "showDocument": {
+                "support": True
+            },
             "showMessage": {
                 "messageActionItem": {
                     "additionalPropertiesSupport": True
                 }
             },
             "workDoneProgress": True
+
```
Comment (rchl) [inline plugin/core/sessions.py:250]: Oh no, no, no, no.

File: plugin/core/sessions.py [hunk 2]
```diff
@@ -1032,6 +1038,21 @@ def m_client_unregisterCapability(self, params: Any, request_id: Any) -> None:
...
+        def success(b: bool) -> None:
+            self.send_response(Response(request_id, {"success": b}))
+
+        if params.get("external"):
+            success(open_externally(uri, bool(params.get("takeFocus"))))
+        else:
+            # TODO: ST API does not allow us to say "do not focus this new view"
+            filename = uri_to_filename(uri)
+            selection = params.get("selection")
+            open_file_and_center_async(self.window, filename, selection).then(lambda _: success(True))
```
Comment (rchl) [inline plugin/core/sessions.py:1054]: Should we take the `open_file_*` take a region right away? Feels weird to pass this poorly typed Dict.

File: plugin/core/open.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    A blocking function that invokes the OS's "open with default extension"
+    """
+    if uri.startswith("http:") or uri.startswith("https:"):
+        return webbrowser.open(uri, autoraise=take_focus)
+    file = uri_to_filename(uri)
+    try:
+        # TODO: handle take_focus
+        if sublime.platform() == "windows":
+            # os.startfile only exists on windows, but pyright does not understand sublime.platform().
+            # TODO: How to make pyright understand platform-specific code with sublime.platform()?
```
Comment (rchl) [inline plugin/core/open.py:77]: I don't know.

**Review by rchl — APPROVED** [review]

File: plugin/core/sessions.py (hunk 2 above)
Comment (rwols) [reply plugin/core/sessions.py:1053]: I pondered on this for a while and I think we should keep using a loose dict. The larger issue is that Range is such a weird class. We have sublime.Region which can already do everything that Range can do. Perhaps we can remove the Range class in the future?

**Pull request merged**